lib/garbage-deleter.js

```diff
@@ -0,0 +1,893 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2020 Joyent, Inc.
+ */
+
+//
+// This contains the code for GarbageDeleter objects which are used by the
+// garbage-deleter program to process garbage collector instructions and delete
+// the appropriate files on disk.
+//
+
+var fs = require('fs');
+var path = require('path');
+var util = require('util');
+
+var assert = require('assert-plus');
+var readline = require('readline');
+var vasync = require('vasync');
+var VError = require('verror').VError;
+
+var common = require('./common');
+
+var DEFAULT_CONCURRENT_DELETES = 10;
+var DEFAULT_MANTA_ROOT = '/manta';
+var DEFAULT_MAX_LINE_LENGTH = 2048;
```

Comment:
We definitely want to guard against junk files being accidentally dropped in. Since we're using node, our memory is easy to outrun.

I can't recall for sure, but I believe 2048 was chosen because it seemed like "more than enough" at the time. Specifically the lines contain:

 * a shardId
 * a storageId
 * two uuids
 * a number (content-length)

which really should never be that large. With the changes that will be made for MANTA-4591, I think we'll end up with at least:

 * a tag (something like "DELETEv2")
 * a storageId
 * a shardId
 * a /manta/v2/<uuid>/<prefix>/<uuid> path
 * a number (content length)

which still seems like it should always easily fit in 2048. Are you concerned about this value being too low?

Reply:
> Are you concerned about this value being too low?

Nope. My only concern is that a comment in the code here could help clarify the reasoning.

Reply:
Added a comment.